detection_rules/mixins.py

```diff
@@ -81,14 +84,39 @@ def dive(child: dict) -> dict:
     return patched
 
 
+class BaseSchema(Schema):
+    """Base schema for marshmallow dataclasses with unknown."""
+    class Meta:
+        unknown = marshmallow.EXCLUDE
+
+
+def exclude_class_schema(clazz, base_schema=BaseSchema, **kwargs):
```

Comment:
Just to double check, `clazz` is not a typo right? This is because in effect you are having an input of class, but class is reserved?